<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="odom_topic" default="/Odometry"/>
  <arg name="robot_name" default="robot0"/>

  

    <node pkg="object_modeller" type="cylinder_plane_modeller.py" name="cylinder_plane_modeller" output="screen" ns="$(arg robot_name)" args="--point_cloud_ns /$(arg robot_name)/">
      <rosparam command="load" file="$(find object_modeller)/config/cylinder_plane_modeller_params.yaml"/>
    </node>
    <node pkg="object_modeller" type="sync_cylinder_odom.py" name="sync_cylinder_odom" output="screen" ns="$(arg robot_name)" args="--odom_topic $(arg odom_topic)">
    </node>
    <node pkg="object_modeller" type="sync_cuboid_odom.py" name="sync_cuboid_odom" output="screen" ns="$(arg robot_name)" args="--odom_topic $(arg odom_topic)">
    </node>
    <node pkg="object_modeller" type="sync_centroid_odom.py" name="sync_centroid_odom" output="screen" ns="$(arg robot_name)" args="--odom_topic $(arg odom_topic)">
    </node>
    <node pkg="object_modeller" type="merge_synced_measurements.py" name="sync_all" output="screen" ns="$(arg robot_name)">
    </node>
</launch>
